credentials/fetcher_default.go

```diff
@@ -124,19 +124,21 @@ func (s *FetcherDefault) fetchParallel(ctx context.Context, locations []url.URL)
 	select {
 	case <-ctx.Done():
 		s.l.Errorf("Ignoring JSON Web Keys from at least one URI because the request timed out waiting for a response.")
+		return errors.Errorf("At least one JWT fetch took more then the allowed duration")
```

Comment:
Checked the docs and passed a few time with debugger, at the end i used `context.DeadlineExceeded` since that gets returned when the timeout is exceeded.